data_browser/src/ExperimentPage.js

```diff
@@ -119,62 +125,322 @@ function renderExperimentHeader(args) {
+  // If the path is under an output path of some step, then link to the name of that step
+  for (const step of steps) {
+    if (path.startsWith(step.output_path)) {
+      return {step, replacedPath: path.replace(step.output_path, `[${step.name}]`)};
+    }
+  }
+  return {step: null, replacedPath: path};
+}
+
+function renderTokenizeStepDescription({step, steps}) {
+  const tokenizer = step.config.tokenizer;
+  const hfUrl = `https://huggingface.co/${tokenizer}/raw/main/tokenizer.json`;
+  const paths = step.config.train_paths.concat(step.config.validation_paths);
+  const links = paths.map((path, index) => {
+    return <div key={index}>{renderPath({path, steps})}</div>;
   });
-  const configTable = configRows.length === 0 ? null :
-    <table className="experiment-step-config"><tbody>{configRows}</tbody></table>;
-
-  // Map dependencies (represented by output paths) to the corresponding step indices
-  const outputPaths = experiment.steps.map(step => step.output_path);
-  const dependencies = step.dependencies.map(dep => `[${outputPaths.indexOf(dep)}]`).join(", ");
-
-  return (<Paper key={index} className="experiment-step">
-    <div>
-      Step [{index}]
-      <span className="description">{step.description}</span>
-    </div>
-    <div className="experiment-step-line"><a href={viewSingleUrl(step.output_path)}>{step.output_path}</a> := {step.fn_name}({dependencies})</div>
-    {configTable}
-    {renderExperimentStatus(events)}
-    <ButtonGroup>
-      {links.map((link, i) => <span key={i}>{link}</span>)}
-    </ButtonGroup>
-  </Paper>);
+  const description = <table>
+    <tbody>
+      <tr><td>Tokenizer:</td><td><a href={hfUrl} target="_blank">{tokenizer}</a></td></tr>
+      <tr><td>Raw text:</td><td>{links}</td></tr>
+    </tbody>
+  </table>;
+  return {name: "tokenize", description};
 }
 
-function renderExperimentStatus(events) {
-  if (!events) {
-    return "[Loading status]";
+function renderTrainStepDescription({step, steps}) {
+  const dataConfig = step.config.train_config.data;
+  const datasetSummary = renderDatasetSummary({dataConfig, steps});
+
+  // Model
+  const modelConfig = step.config.train_config.model;
+  const architectureSummary = [
+    `d_model = ${modelConfig.hidden_dim}`,
+    `d_ff = ${modelConfig.intermediate_dim}`,
+    `n_heads = ${modelConfig.num_heads} (${modelConfig.num_kv_heads} kv)`,
+    `n_layers = ${modelConfig.num_layers}`,
+    `seq_len = ${modelConfig.seq_len}`,
+    `rope(${modelConfig.rope.factor},${modelConfig.rope.theta})`,
+    modelConfig.activation_function,
+  ].join(", ");
+
+  // Optimizer
+  const optimizerConfig = step.config.train_config.optimizer;
+  const finalLearningRate = optimizerConfig.learning_rate * optimizerConfig.min_lr_ratio;
+  const optimizerSummary = [
+    `lr = ${renderSciNotation(optimizerConfig.learning_rate)} → ${renderSciNotation(finalLearningRate)} (${optimizerConfig.lr_schedule})`,
+    `warmup = ${optimizerConfig.warmup}`,
```

Comment:
cooldown and LR schedule type is pretty important

Reply:
I think `lr_schedule` provides the type (linear, etc) or do you mean something else?

Reply:
oh missed that. but you do need to specify cooldown for e.g. WSD

Reply:
Hmm, I have init lr -> final lr - how should I display the cooldown?